crates/router/src/core/refunds.rs

```diff
@@ -634,12 +634,26 @@ pub async fn refund_list(
     req: api_models::refunds::RefundListRequest,
 ) -> RouterResponse<api_models::refunds::RefundListResponse> {
     let limit = validator::validate_refund_list(req.limit)?;
+    let offset = validator::validate_refund_offset(req.offset)?;
+
     let refund_list = db
         .filter_refund_by_constraints(
             &merchant_account.merchant_id,
             &req,
             merchant_account.storage_scheme,
             limit,
+            offset,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::RefundNotFound)?;
```

Comment:
```suggestion
        .to_not_found_response(errors::ApiErrorResponse::RefundNotFound)?;
```

Reply:
This should only return RefundNotFound if the DB returned a NotFound error for other critical failures this could be internal_server_error